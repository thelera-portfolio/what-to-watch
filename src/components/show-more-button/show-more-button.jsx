import {ActionCreator} from "../../reducer/data/data.js";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import React from "react";
import {SHOWING_FILMS_COUNT} from "../../utils/consts.js";

const ShowMoreButton = (props) =>
  <button
    className="catalog__button"
    type="button"
    onClick={() => props.onClick(SHOWING_FILMS_COUNT)}>Show more
  </button>;

ShowMoreButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onClick(count) {
    dispatch(ActionCreator.incrementFilmsCount(count));
  },
});

export {ShowMoreButton};
export default connect(null, mapDispatchToProps)(ShowMoreButton);
