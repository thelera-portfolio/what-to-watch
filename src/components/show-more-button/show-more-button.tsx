import * as React from "react";
import {ActionCreator} from "../../reducer/data/data";
import {connect} from "react-redux";
import {SHOWING_FILMS_COUNT} from "../../utils/consts";

interface Props {
  onClick: () => void,
}

const ShowMoreButton: React.FunctionComponent<Props> = (props: Props) => {
  const {onClick} = props;

  return (
    <button
      className="catalog__button"
      type="button"
      onClick={onClick}
    >
      Show more
    </button>
  );
};

const mapDispatchToProps = (dispatch) => ({
  onClick() {
    dispatch(ActionCreator.incrementFilmsCount(SHOWING_FILMS_COUNT));
  },
});

export {ShowMoreButton};
export default connect(null, mapDispatchToProps)(ShowMoreButton);
