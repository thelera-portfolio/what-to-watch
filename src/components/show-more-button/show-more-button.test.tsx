import * as React from "react";
import * as renderer from "react-test-renderer";
import {ShowMoreButton} from "./show-more-button";

it(`ShowMoreButton is rendered correctly`, () => {
  const tree = renderer.create(
      <ShowMoreButton
        onClick={() => null}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
