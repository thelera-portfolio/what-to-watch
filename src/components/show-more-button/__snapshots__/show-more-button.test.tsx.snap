// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreButton is rendered correctly 1`] = `
<button
  className="catalog__button"
  onClick={[Function]}
  type="button"
>
  Show more
</button>
`;
